plugins/chain_plugin/include/eosio/chain_plugin/blockvault_sync_strategy.hpp

```diff
@@ -84,7 +84,7 @@ struct blockvault_sync_strategy : public sync_callback {
 
       try {
          ++_num_blocks_received;
-         EOS_ASSERT(_blockchain_provider.incoming_blockvault_sync_method(block), plugin_exception,
+         EOS_ASSERT(_blockchain_provider.incoming_blockvault_sync_method(block, (1 != _num_blocks_received)), plugin_exception,
```

Comment:
I don’t think we can assume it’s only the first block that needs to skip the connectivity check.  We still have the edge case of a snapshot that sliced 2 or more possible chains.  We cannot guarantee that the first block from BV is on the best eventual chain.  

Can this be based on block height?  

If the incoming block is exactly snapshot height + 1 then we cannot perform the connectivity check.